### PR TITLE
Fix various issues in location system mode calls

### PR DESCRIPTION
**Pull request by dmulcahey**
**Describe what the PR does:**
This PR corrects the value for the location calls that set system modes.
  
**Checklist:**

- [X] Confirm that one or more new tests are written for the new functionality. - existing tests updated
- [X] Update `README.md` with any new documentation. - no functionality added
- [X] Add yourself to `AUTHORS.md`.

### CODE REVIEW

**Comment by dmulcahey** [comment]
Thoughts on this @bachya? Is there a way to make the lib used for mocking the calls validate the request data? I am not familiar with the lib at all.

**Comment by bachya** [comment]
@dmulcahey Can you give me more info? Are you wanting your tests to blow up on bad input?

**Comment by dmulcahey** [comment]
@bachya these tests were passing before. In the lib in HA the mock also gets set up with the posted payload... ex: `json={“some”:”payload”}` and it will only match the request if the entire payload matches.

**Comment by dmulcahey** [comment]
@bachya Think I got it. I updated the responses lib and I updated the tests. If this is good w/ you can you release this again after this is merged? Thanks!

File: tests/test_location.py [hunk 1]
```diff
@@ -68,18 +68,21 @@ async def test_system_modes(aresponses, auth_success_response):
...
         "post",
         aresponses.Response(text=None, status=204),
+        body_pattern='{"target": "away"}',
     )
     aresponses.add(
         "api-gw.meetflo.com",
         "/api/v2/locations/mmnnoopp/systemMode",
         "post",
         aresponses.Response(text=None, status=204),
+        body_pattern='{"target": "home"}',
```
Comment (bachya) [inline tests/test_location.py:78]: Is this the entire body that you expect? If so, the normal way to do it in `aresponses` is the via the `text` kwarg of the `aresponses.Response` class (similar to line 77). That parameter just takes text that represents the response body.

**Pull request merged**